### PR TITLE
Feat/heatmap

### PR DESCRIPTION
**Pull request by Ollekmc**
Add heatMap pls review

### CODE REVIEW

File: frontend/src/component/UserOverview.tsx [hunk 1]
```diff
@@ -19,12 +20,17 @@ export default function UserOverview(props:UserOverviewProps) {
...
+    }
+
     return (<div>
         <UserList
             users={users}
             handleLoginAs={handleLoginAs}
             deleteUser={handleDeleteUser}
             loggedInUser={props.loggedInUser}/>
         Logged in as : {props.loggedInUser && props.loggedInUser.name}
+        <button onClick={openMap}>map </button>
```
Comment (fannie-id) [inline frontend/src/component/UserOverview.tsx:34]: `<button onClick={openMap}> map </button>`

**Review by fannie-id — APPROVED** [review]

File: frontend/src/component/HeatMap.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        <ComposableMap
+            projectionConfig={{
+                rotate: [-10, 0, 0],
+                scale: 147
+            }}
+        >
+            {countryCounts.length > 0 && (
+                <Geographies geography={geoUrl}>
+                    {({geographies}) =>
+                        geographies.map((geo) => {
```
Comment (NickKlock) [inline frontend/src/component/HeatMap.tsx:65]: arrow functions always return a value.

File: frontend/src/component/HeatMap.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        .domain([0, maxDomain])
+        .range(["#f8bbd0", "#ad1457"]);
+
+    const countryList = props.countries
+    let userList:Traveller[] = useUsers().users
+
+    const [countryCounts, setCountryCounts] = useState<{
+        country: string,
+        count: number
+    }[]>([])
```
Comment (NickKlock) [inline frontend/src/component/HeatMap.tsx:27]: you can create a type for that

File: frontend/src/component/HeatMap.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                    element.count++
+                }
+            }
+        }
+        let countVisits = []
+        for (const element of output) countVisits.push(element.count)
+        let maxVisits = Math.max(...countVisits)
+        setMaxDomain(maxVisits)
+        setCountryCounts(output)
+    }, [])
```
Comment (NickKlock) [inline frontend/src/component/HeatMap.tsx:53]: Someone else then your probably needs a day to understand your code here. maybe less for-each and more filter(), map() etc?

**Review by NickKlock — CHANGES_REQUESTED** [review]

File: frontend/src/component/HeatMap.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        const threeLetterCodeList = countryList.map(Country => Country.threeLetterCode)
+
+        function setCount(country: string) {
+            return {
+                country,
+                count: 0
+            };
+        }
+
+        let output = threeLetterCodeList.map(setCount);
```
Comment (NickKlock) [inline frontend/src/component/HeatMap.tsx:39]: what is output suppose to mean? maybe u can be a bit more creative with your variable names.

**Review by NickKlock — APPROVED** [review]

**Pull request merged**